From 480d6355987f3b549e12f51250b8de7cf4993050 Mon Sep 17 00:00:00 2001
From: retro98boy <retro98boy@qq.com>
Date: Mon, 8 Apr 2024 10:46:02 +0800
Subject: [PATCH 06/10] Add display kmod packages about Rockchip

---
 package/kernel/linux/modules/video.mk | 54 +++++++++++++++++++++++++++
 1 file changed, 54 insertions(+)

diff --git a/package/kernel/linux/modules/video.mk b/package/kernel/linux/modules/video.mk
index f1e61f3..0b00a1e 100644
--- a/package/kernel/linux/modules/video.mk
+++ b/package/kernel/linux/modules/video.mk
@@ -386,6 +386,60 @@ endef
 
 $(eval $(call KernelPackage,drm-imx-ldb))
 
+
+define KernelPackage/drm-rockchip
+  SUBMENU:=$(VIDEO_MENU)
+  TITLE:=RockChip DRM support
+  DEPENDS:=@TARGET_rockchip +kmod-drm-kms-helper +kmod-sound-soc-core
+  KCONFIG:=CONFIG_DRM_ROCKCHIP \
+  CONFIG_DRM_FBDEV_EMULATION=y \
+  CONFIG_DRM_FBDEV_OVERALLOC=100 \
+  CONFIG_ROCKCHIP_DW_HDMI=y \
+  CONFIG_DRM_DW_HDMI_AHB_AUDIO \
+  CONFIG_DRM_DW_HDMI_I2S_AUDIO \
+  CONFIG_DRM_DW_HDMI_CEC \
+  CONFIG_SND_SOC_HDMI_CODEC \
+  CONFIG_ROCKCHIP_DW_MIPI_DSI=y \
+  CONFIG_DRM_PANEL=y \
+  CONFIG_DRM_PANEL_TOSHIBA_TC358775
+  FILES:= \
+  $(LINUX_DIR)/drivers/gpu/drm/rockchip/rockchipdrm.ko \
+  $(LINUX_DIR)/drivers/gpu/drm/bridge/synopsys/dw-hdmi.ko \
+  $(LINUX_DIR)/drivers/gpu/drm/bridge/synopsys/dw-hdmi-ahb-audio.ko \
+  $(LINUX_DIR)/drivers/gpu/drm/bridge/synopsys/dw-hdmi-i2s-audio.ko \
+  $(LINUX_DIR)/drivers/gpu/drm/bridge/synopsys/dw-hdmi-cec.ko \
+  $(LINUX_DIR)/drivers/media/cec/core/cec.ko \
+  $(LINUX_DIR)/sound/soc/codecs/snd-soc-hdmi-codec.ko \
+  $(LINUX_DIR)/drivers/gpu/drm/bridge/synopsys/dw-mipi-dsi.ko \
+  $(LINUX_DIR)/drivers/gpu/drm/panel/panel-toshiba-tc358775.ko
+  AUTOLOAD:=$(call AutoLoad,08,rockchipdrm dw-hdmi dw-hdmi-ahb-audio dw-hdmi-i2s-audio dw-hdmi-cec cec snd-soc-hdmi-codec dw-mipi-dsi panel-toshiba-tc358775)
+endef
+
+define KernelPackage/drm-rockchip/description
+  Direct Rendering Manager (DRM) support for RockChip
+endef
+
+$(eval $(call KernelPackage,drm-rockchip))
+
+
+define KernelPackage/drm-panfrost
+  SUBMENU:=$(VIDEO_MENU)
+  TITLE:=ARM Mali Midgard/Bifrost GPUs DRM support
+  DEPENDS:=+kmod-drm-kms-helper
+  KCONFIG:=CONFIG_DRM_PANFROST
+  FILES:= \
+  $(LINUX_DIR)/drivers/gpu/drm/panfrost/panfrost.ko \
+  $(LINUX_DIR)/drivers/gpu/drm/scheduler/gpu-sched.ko
+  AUTOLOAD:=$(call AutoLoad,08,panfrost gpu-sched)
+endef
+
+define KernelPackage/drm-panfrost/description
+  Direct Rendering Manager (DRM) support for ARM Mali Midgard/Bifrost GPUs
+endef
+
+$(eval $(call KernelPackage,drm-panfrost))
+
+
 define KernelPackage/drm-radeon
   SUBMENU:=$(VIDEO_MENU)
   TITLE:=Radeon DRM support
-- 
2.44.0

